_stbt/core.py

```diff
@@ -1189,8 +1191,22 @@ def match_text(self, text, frame=None, region=Region.ALL,
         return result
 
     def frames(self, timeout_secs=None):
-        for f in self._display.frames(timeout_secs):
-            yield f.copy(), int(f.time * 1e9)
+        if timeout_secs is not None:
+            end_time = self._time.time() + timeout_secs
+        timestamp = None
+
+        while True:
+            ddebug("user thread: Getting sample at %s" % self._time.time())
+            frame = self._display.get_frame(
+                max(10, timeout_secs or 0), since=timestamp)
+            ddebug("user thread: Got sample at %s" % self._time.time())
+            timestamp = frame.time
+
+            yield frame.copy(), int(f.time * 1e9)
```

Comment:
`f.time` should be `frame.time`; or better, `timestamp`.

Reply:
I think I fix this later in the series.  I'll consider fixing it in the rebase.

Reply:
Fixed in rebase